### ISSUE TITLE
Wrong CMAKE_VERSION check for Foundation

### ISSUE THREAD

**Issue by avovana** [pocoproject/poco#2668]
#### Expected behavior
compile
#### Actual behavior
error
#### Steps to reproduce the problem
`cmake .`
-- Poco package version: 2.0.0
-- Checking for C++14 compiler
-- Checking for C++14 compiler - available
-- Building dynamic libraries
-- Could NOT find APR (missing:  APR_INCLUDE_DIR APR_LIBRARY) 
-- Could NOT find Apache2 (missing:  APACHE2_INCLUDE_DIR) 
-- Could NOT find MYSQL (missing:  MYSQL_INCLUDE_DIR MYSQL_LIBRARY) 
-- Could NOT find PostgreSQL (missing:  PostgreSQL_LIBRARY PostgreSQL_INCLUDE_DIR) 
-- Could NOT find ODBC (missing:  ODBC_LIBRARY ODBC_INCLUDE_DIR) 
-- Building without tests
-- Building without samples
-- Build with using internal copy of sqlite, libz, pcre, expat, ...
CMake Error at Foundation/CMakeLists.txt:118 (target_compile_features):
  **target_compile_features specified unknown feature "cxx_std_14" for target
  "Foundation".**

-- CMake 3.7.2 successfully configured Poco using Unix Makefiles generator
-- Installation target path: /usr/local
-- Generated with config types: 
-- [cmake] build type:             RelWithDebInfo
-- [cmake] build with cxx flags:   -O2 -g -DNDEBUG 
-- [cmake] build with c flags:     -O2 -g -DNDEBUG 
-- Building: CppUnit
-- Building: Encodings
-- Building: XML
-- Building: JSON
-- Building: Util
-- Building: Net
-- Building: MongoDB
-- Building: Redis
-- Building: Crypto
-- Building: NetSSL_OpenSSL
-- Building: Zip
-- Building: PageCompiler
-- Building: File2Page
-- Configuring incomplete, errors occurred!
See also "/home/user/tmp/poco-develop/CMakeFiles/CMakeOutput.log".
See also "/home/user/tmp/poco-develop/CMakeFiles/CMakeError.log".

#### POCO version
poco-1.9.0-release
#### Compiler and version
gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
cmake version 3.7.2
#### Operating system and version
Debian 9.5
#### Other relevant information
Here is Foundation/CMakeLists.txt:

> if(NOT DISABLE_CPP14 AND CMAKE_VERSION VERSION_GREATER 3.7)
>   target_compile_features(Foundation PUBLIC cxx_std_14)
> endif()

As far as I know `cxx_std_YY` feature [appeared since CMAKE 3.8](https://crascit.com/2015/03/28/enabling-cxx11-in-cmake/).

I fixed it with this code:

> set(CMAKE_CXX_STANDARD 14)
> set(CMAKE_CXX_STANDARD_REQUIRED ON)
> ...
> if(NOT DISABLE_CPP14 AND CMAKE_VERSION VERSION_GREATER 3.8)
>   target_compile_features(Foundation PUBLIC cxx_std_14)
> endif()

But I'm making now automation process for downloading, unpacking, installing on every device I need. So I can not make such fix every time.
Can you fix it?

P.S.
Or it maybe solved something like this:

> if(NOT DISABLE_CPP14 AND CMAKE_VERSION VERSION_GREATER 3.8)
>    target_compile_features(Foundation PUBLIC cxx_std_14)
>    else if(NOT DISABLE_CPP14 AND CMAKE_VERSION VERSION_GREATER 3.1)
>        set_target_properties(Foundation PROPERTIES
>        CXX_STANDARD 14
>        CXX_STANDARD_REQUIRED YES
>        )
>        endif()
> endif()

**Comment by Bjoe** [comment]
Hi. Sorry my mistake ... I was expected that VERSION_GREATER 3.7 only check the major and minor version ... I not realized that they include also the patch version number. 
To set the Foundation target properties `CXX_STANDARD` and `CXX_STANDARD_REQUIRED` is not required because this are set in the "root" CMakeLists.txt.
I created a pull request/branch #2671 . If you can test this, would be great.
I added this `target_compile_features(Foundation PUBLIC cxx_std_14)` if I cleanup/refactor the cmake build system but I'm even wondering if this is still needed ?

**Comment by avovana** [comment]
@Bjoe Thanks for a reply. It helped me to understand VERSION_GREATER better.
At this moment I decided to use the last stable version of CMake(3.14). It compiled.

About testing with CMake 3.7.2. I tested yesterday it with VERSION_GREATER 8, but with `set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED ON)` too. It compiled.

**Comment by Bjoe** [comment]
Hi. I merged my Fix. Can you please test and close this issue if it works. Thanks. 
Btw. I also fixed `poco-1.9.1` branch.

**Comment by MarcoSpeziali** [comment]
I actually have the same problem with the latest version (from conan):

------------------

**Expected behavior**

compile

**Actual behavior**

error

**Steps to reproduce the problem**

`cmake .`

```
poco/1.10.1: Copying sources to build folder
poco/1.10.1: Building your package in /Users/Marco/.conan/data/poco/1.10.1/_/_/build/231e22a6feb2d43c7aa1de4ee84a3456f197dedf
poco/1.10.1: Generator cmake created conanbuildinfo.cmake
poco/1.10.1: Calling build()
poco/1.10.1: OrderedDict([('CMAKE_BUILD_TYPE', 'Debug'), ('CONAN_IN_LOCAL_CACHE', 'ON'), ('CONAN_COMPILER', 'clang'), ('CONAN_COMPILER_VERSION', '10'), ('CONAN_CXX_FLAGS', '-m64'), ('CONAN_SHARED_LINKER_FLAGS', '-m64'), ('CONAN_C_FLAGS', '-m64'), ('CONAN_LIBCXX', 'libc++'), ('BUILD_SHARED_LIBS', 'OFF'), ('CMAKE_INSTALL_PREFIX', '/Users/Marco/.conan/data/poco/1.10.1/_/_/package/231e22a6feb2d43c7aa1de4ee84a3456f197dedf'), ('CMAKE_INSTALL_BINDIR', 'bin'), ('CMAKE_INSTALL_SBINDIR', 'bin'), ('CMAKE_INSTALL_LIBEXECDIR', 'bin'), ('CMAKE_INSTALL_LIBDIR', 'lib'), ('CMAKE_INSTALL_INCLUDEDIR', 'include'), ('CMAKE_INSTALL_OLDINCLUDEDIR', 'include'), ('CMAKE_INSTALL_DATAROOTDIR', 'share'), ('CONAN_CMAKE_POSITION_INDEPENDENT_CODE', 'ON'), ('CMAKE_EXPORT_NO_PACKAGE_REGISTRY', 'ON'), ('CONAN_EXPORTED', '1'), ('CXX_14', 'OFF'), ('ENABLE_APACHECONNECTOR', 'OFF'), ('ENABLE_CPPPARSER', 'OFF'), ('ENABLE_CRYPTO', 'ON'), ('ENABLE_DATA', 'ON'), ('ENABLE_DATA_MYSQL', 'OFF'), ('ENABLE_DATA_ODBC', 'OFF'), ('ENABLE_DATA_POSTGRESQL', 'OFF'), ('ENABLE_DATA_SQLITE', 'ON'), ('ENABLE_ENCODINGS', 'ON'), ('ENABLE_JSON', 'ON'), ('ENABLE_JWT', 'ON'), ('ENABLE_MONGODB', 'ON'), ('ENABLE_NET', 'ON'), ('ENABLE_NETSSL', 'ON'), ('ENABLE_NETSSL_WIN', 'ON'), ('ENABLE_PAGECOMPILER', 'OFF'), ('ENABLE_PAGECOMPILER_FILE2PAGE', 'OFF'), ('ENABLE_PDF', 'OFF'), ('ENABLE_POCODOC', 'OFF'), ('ENABLE_REDIS', 'ON'), ('ENABLE_SEVENZIP', 'OFF'), ('ENABLE_TESTS', 'OFF'), ('ENABLE_UTIL', 'ON'), ('ENABLE_XML', 'ON'), ('ENABLE_ZIP', 'ON'), ('FORCE_OPENSSL', 'ON'), ('POCO_UNBUNDLED', 'OFF'), ('POCO_STATIC', 'ON'), ('CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP', True)])
-- The C compiler identification is Clang 10.0.0
-- The CXX compiler identification is Clang 10.0.0
-- Check for working C compiler: /usr/local/opt/llvm/bin/clang
-- Check for working C compiler: /usr/local/opt/llvm/bin/clang - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/local/opt/llvm/bin/clang++
-- Check for working CXX compiler: /usr/local/opt/llvm/bin/clang++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan CMake Wrapper
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Conan: Adjusting fPIC flag (ON)
-- Conan: Compiler Clang>=8, checking major version 10
-- Conan: Checking correct version: 10
-- Conan: C++ stdlib: libc++
-- Checking for C++14 compiler
-- Checking for C++14 compiler - available
CMake Warning at CMakeListsOriginal.cmake:57 (message):
  POCO_STATIC has been deprecated.  Please use BUILD_SHARED_LIBS instead.
Call Stack (most recent call first):
  CMakeLists.txt:11 (include)


-- Found OpenSSL: /Users/Marco/.conan/data/openssl/1.1.1g/_/_/package/95b8985581387a30feb3d9d93046ec9febbf4d45/lib/libcrypto.a (found version "1.1.1g")  
-- Found APR: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/apr-1 (found version "1.5.2") 
-- Found APRUTIL: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/apr-1 (found version "1.5.4") 
-- Could NOT find Apache2 (missing: APACHE2_INCLUDE_DIR) 
-- Could NOT find MYSQL (missing: MYSQL_INCLUDE_DIR MYSQL_LIBRARY) 
-- Could NOT find PostgreSQL (missing: PostgreSQL_INCLUDE_DIR) 
-- Could NOT find ODBC (missing: ODBC_INCLUDE_DIR) 
-- Building without tests & samples
-- Using internal sqlite, zlib, pcre, expat, ...
CMake Error at Foundation/CMakeLists.txt:128 (target_compile_features):
  target_compile_features no known features for CXX compiler

  "Clang"

  version 10.0.0.


CMake Error at JSON/CMakeLists.txt:38 (target_compile_features):
  target_compile_features no known features for C compiler

  "Clang"

  version 10.0.0.
```

**POCO version**

`poco-1.10.1`

**Compiler and version**

`clang version 10.0.0 (x86_64-apple-darwin19.6.0)`
`cmake 3.16.5`

**Operating system and version**

`macOS 10.15.6`
